eclair-core/src/main/scala/fr/acinq/eclair/message/Postman.scala

```diff
@@ -88,10 +88,10 @@ object Postman {
           Behaviors.same
         case SendMessage(intermediateNodes, destination, replyPath, messageContent, replyTo, timeout) =>
           val messageId = randomBytes32()
-          val replyRoute = replyPath.map(replyHops => {
+          val replyRoute = replyPath.flatMap(replyHops => {
             val intermediateHops = replyHops.dropRight(1).map(OnionMessages.IntermediateNode(_))
             val lastHop = OnionMessages.Recipient(replyHops.last, Some(messageId))
-            OnionMessages.buildRoute(randomKey(), intermediateHops, lastHop).get
+            OnionMessages.buildRoute(nodeParams.privateKey, randomKey(), intermediateHops, lastHop).toOption
```

Comment:
What if we want the reply path to be zero hop? This breaks it.

Reply:
But...that's completely dumb, isn't it? Even if you want the receiver to send the response directly to you, it's always preferable to use a blinded path from them to you, which gives deniability?

Reply:
This change disappeared in https://github.com/ACINQ/eclair/pull/2589/commits/9dd1f2e822a4563cefd66618326da537cdf2b612